등록 컴포넌트

yejineee opened the pull request:
- styled-components 로 스타일 설정하였습니다.
- index.html을 public으로 옮긴 후, 웹팩에서 경로 설정을 바꿔주었습니다.
- 등록 컴포넌트를 생성해보았습니다.

- 궁금한 점...
  - 함수 이름을 NewButton으로 하고, html tag를 StyledButton으로 하였는데, 
그 반대로 해야 하는건지 궁금합니다...!

Commit:
Chore: index.html 위치 변경 및 프론트 웹팩 경로 변경

프론트엔드 폴더 구조를 설계하는 과정에서 public폴더를 만들었고,
index.html을 정적 파일들이 들어있는 public 폴더로 옮겼다.
그에 맞춰서 webpack의 index.html의 경로도 수정하였다.

Commit:
Feat: 등록 버튼 컴포넌트 추가

마일스톤, 이슈, 라벨 생성 시 사용되는 버튼 컴포넌트를 추가하였다.
handleClick을 prop으로 받아서, 클릭 시 다른 페이지로 넘어가도록 하였다.

Commit:
Chore: styled-components, prop-types 설치

style-in-js 라이브러리 중 가장 인기있는 styled-components를 설치하였다.
props에서 validate 검사를 하지 않으면, eslint error가 발생하여
prop-types를 설치하여, props의 타입을 확인하도록 하였다.

changheedev commented on front/src/components/NewButton.js:
```diff
@@ -0,0 +1,26 @@
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const StyledButton = styled.button`
```
newButton의 스타일이니까 newButtonStyle 같은 이름이 더 어울리지 않을까요??
다른분들의 의견도 들어보겠습니다 👀

yejineee replied:
좀 더 구체적인 이름으로 바꿔보겠습니다!

rolled-potatoes commented on front/src/components/NewButton.js:
```diff
@@ -0,0 +1,26 @@
+  font-size: 1rem;
+`;
+
+function NewButton(props) {
+  const { title, handleClick } = props;
```
매개 변수를 객체로 받는게 좋을 것 같아요
```js
function NewButton({ title, handleClick } ) {
```
그리고 컴포넌트 안에 들어갈 내용을 
```js
// 예시
<NewButton>title</NewButton>
function NewButton({handleClick,children} ) {
  return <StyledButton onClick={handleClick}>{children}</StyledButton>;
}
```
children을 이용하면 자식으로 들어간 내용을 불러 올 수 있어요

yejineee replied:
아하 children 좀 더 공부해서 적용해볼게요! 감사합니다~

rolled-potatoes commented on front/src/components/NewButton.js:
```diff
@@ -0,0 +1,26 @@
+  border-radius: 0.4rem;
+
+  background: #2da44f;
+  color: white;
```
색상같은 경우는 두루두루 쓰일 수도 있고
서로 맞춰야하는 부분일 것 같은데 따로 js 파일로 분리해서 string으로 지정 해두는게 좋을 것 같습니다.

yejineee replied:
아 네네! 객체로 따로 지정해두겠습니다!

Commit:
FEAT: props.children 사용 & 태그 이름 변화

props.children을 사용하여 컴포넌트에서 다른 컴포넌트를 담을 수 있도록 하였다.
태그 이름을 StyledButton에서 좀 더 구체적인 이름으로 변경하였다.

Commit:
Feat: css color object file 추가

여러 파일에서 사용할 컬러들의 번호를 객체로 묶어서 저장한다.
이를 통해 다른 파일에서 동일한 색을 사용할 수 있고,
수정도 용이하게 한다.

changheedev commented on front/src/components/NewButton.js:
```diff
@@ -0,0 +1,26 @@
+`;
+
+function NewButton({ handleClick, children }) {
+  return <StyledNewButton onClick={handleClick}>{children}</StyledNewButton>;
+}
```
아 단순히 스타일을 담아두는 변수인줄 알고 newButtonStyle 같이 말씀 드렸던건데 그게 아니라 스타일을 지정한 컴포넌트를 생성하는거였네요. 🤔

yejineee replied:
네넹..! 같은 이름으로 지정해줘야 스타일이 적용되더라구욤!
